test/rules/unnecessary_this.dart

```diff
@@ -4,6 +4,32 @@
 
 // test w/ `pub run test -N unnecessary_this`
 
+void bar() {
+
+}
+
+int x;
+
+
+class Parent {
+  int x, y;
+  void bar() {
+
+  }
+}
+
+class Child extends Parent {
+  void foo() {
+    this.bar(); // LINT
+  }
+  void baz(int a) {
+    this.x = a; // LINT
```

Comment:
This is similarly wrong. Without the `this`, `x` will refer to the top-level variable.